### PR TITLE
 [function](string_function) add new string function 'extract_url_parameter'

### PR DESCRIPTION
**Pull request by DongLiang-0**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.
[function](string_function) add new string function 'extract_url_parameter'

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by HappenLee** [comment]
Rethink the right way of the function
```
select extractURLParameter('https://localhost:3000?id=abcdefghi', 'id');

SELECT extractURLParameter('https://localhost:3000?id=abcdefghi', 'id')

Query id: 3301471c-ea38-4792-b5d0-bf6b98d0bf05

┌─extractURLParameter('https://localhost:3000?id=abcdefghi', 'id')─┐
│ abcdefghi                                                        │
└──────────────────────────────────────────────────────────────────┘

1 row in set. Elapsed: 0.001 sec. 
```

**Comment by DongLiang-0** [comment]
> 

it's done!

PTAL

**Comment by DongLiang-0** [comment]
> Rethink the right way of the function
> 
> ```
> select extractURLParameter('https://localhost:3000?id=abcdefghi', 'id');
> 
> SELECT extractURLParameter('https://localhost:3000?id=abcdefghi', 'id')
> 
> Query id: 3301471c-ea38-4792-b5d0-bf6b98d0bf05
> 
> ┌─extractURLParameter('https://localhost:3000?id=abcdefghi', 'id')─┐
> │ abcdefghi                                                        │
> └──────────────────────────────────────────────────────────────────┘
> 
> 1 row in set. Elapsed: 0.001 sec. 
> ```

it's done!
PTAL @HappenLee

**Review by zhangstar333 — COMMENTED** [review]
small problems, could update code to fixed

File: be/src/util/url_parser.cpp [hunk 1]
```diff
@@ -344,4 +346,59 @@ UrlParser::UrlPart UrlParser::get_url_part(const StringValue& part) {
     }
 }
 
+std::string UrlParser::extract_url(const StringValue& url, const StringValue& name) {
```
Comment (HappenLee) [inline be/src/util/url_parser.cpp:349]: replace all `std::string` by `std::string_view` to speed up the function.

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 40.19 seconds
 load time: 583 seconds
 storage size: 17154768716 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221019201634_clickbench_pr_31425.html

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**